pycarol/cds.py

```diff
@@ -370,7 +370,49 @@ def sync_data(self, dm_name, dm_id=None, num_records=-1, file_pattern='*', filte
         return self.carol.call_api(path='v1/cds/golden/fetchData', method='POST', params=query_params,
                                    data=filter_query)
 
-    def delete(self, dm_name=None, dm_id=None):
+
+    def delete_rejected(self, dm_name=None, dm_id=None, connector_name=None, connector_id=None, staging_name=None):
+        """
+        Delete CDS DataModel rejected data
+
+        Args:
+            dm_name: `str`,
+                Data Model name.
+            dm_id: `str`, default `None`
+                Data Model id.
+            connector_name: `str`, default `None`
+                Connector name. Used if delete only records from a given connector/staging table
+            connector_id:  `str`, default `None`
+                Connector id. Used if delete only records from a given connector/staging table
+            staging_name:  `str`, default `None`
+                Staging name. Used if delete only records from a given connector/staging table
+
+        Returns: dict
+            Carol task created.
+
+
+        """
+
+
+        if dm_name:
+            dm_id = DataModel(self.carol).get_by_name(dm_name)['mdmId']
+        else:
+            if dm_id is None:
+                raise ValueError('dm_name or dm_id should be set.')
+
+        if connector_name:
+            connector_id = Connectors(self.carol).get_by_name(connector_name)['mdmId']
+
```

Comment:
Shouldn't have a validation to check whether the connector id is empty or not in case no connector name is provided?

Reply:
if not connector providade it will delete all rejected. This is only used if one wants to delete a the rejected records that came from given staging. 

Actually, we need to validatee if Staging name is not None, I need to have a connector.. I will add.

Reply:
done